test/with_dummyserver/test_https.py

```diff
@@ -67,6 +70,34 @@ def test_set_ssl_version_to_tlsv1(self):
         r = self._pool.request('GET', '/')
         self.assertEqual(r.status, 200, r.data)
 
+    def test_client_intermediate(self):
+        client_cert, client_key, client_subject = (
+            DEFAULT_CLIENT_CERTS['certfile'],
+            DEFAULT_CLIENT_CERTS['keyfile'],
+            DEFAULT_CLIENT_CERTS['subject']
+        )
+        https_pool = HTTPSConnectionPool(self.host, self.port,
+                                         key_file=client_key,
+                                         cert_file=client_cert)
+        r = https_pool.request('GET', '/certificate')
+        self.assertDictEqual(json.loads(r.data.decode('utf-8')),
+                             client_subject, r.data)
+
+    def test_client_no_intermediate(self):
+        client_cert, client_key = (
+            DEFAULT_CLIENT_NO_INTERMEDIATE_CERTS['certfile'],
+            DEFAULT_CLIENT_NO_INTERMEDIATE_CERTS['keyfile']
+        )
+        https_pool = HTTPSConnectionPool(self.host, self.port,
+                                         cert_file=client_cert,
+                                         key_file=client_key)
+        try:
+            https_pool.request('GET', '/certificate', retries=False)
+        except SSLError as e:
+            self.assertTrue('alert unknown ca' in str(e) or
+                            'invalid certificate chain' in str(e) or
+                            'unknown Cert Authority' in str(e))
```

Comment:
Nit: should we skip these two tests if `CERT_OPTIONAL` is `None`?

Reply:
I just opened a PR (https://github.com/shazow/urllib3/pull/1251) and I had the travis build fail on a similar assert, but not this line:
```
    def test_client_no_intermediate(self):
        client_cert, client_key = (
            DEFAULT_CLIENT_NO_INTERMEDIATE_CERTS['certfile'],
            DEFAULT_CLIENT_NO_INTERMEDIATE_CERTS['keyfile']
        )
        https_pool = HTTPSConnectionPool(self.host, self.port,
                                         cert_file=client_cert,
                                         key_file=client_key)
        try:
            https_pool.request('GET', '/certificate', retries=False)
        except SSLError as e:
            self.assertTrue('alert unknown ca' in str(e) or
                            'invalid certificate chain' in str(e) or
>                           'unknown Cert Authority' in str(e))
E           AssertionError: False is not true
test/with_dummyserver/test_https.py:99: AssertionError
```
Could my failure be related to this commit?

Reply:
Nope, your failure is just a transient one caused by Tornado misbehaving. =)